packages/scene-composer/src/components/panels/SceneHierarchyPanel/SceneHierarchyDataProvider.tsx

```diff
@@ -193,44 +190,41 @@ const SceneHierarchyDataProvider: FC<SceneHierarchyDataProviderProps> = ({ selec
     (objectRef: string, newParentRef?: string) => {
       updateSceneNodeInternal(objectRef, { parentRef: newParentRef });
     },
-    [sceneComposerId],
+    [updateSceneNodeInternal, getSceneNodeByRef, nodeMap],
   );
 
   const show = useCallback(
     (objectRef: string) => {
-      const getObject3DBySceneNodeRef = useStore(sceneComposerId).getState().getObject3DBySceneNodeRef;
       const object = getObject3DBySceneNodeRef(objectRef);
       if (object) {
         object.visible = true;
       }
     },
-    [sceneComposerId],
+    [getObject3DBySceneNodeRef],
   );
 
   const hide = useCallback(
     (objectRef: string) => {
-      const getObject3DBySceneNodeRef = useStore(sceneComposerId).getState().getObject3DBySceneNodeRef;
       const object = getObject3DBySceneNodeRef(objectRef);
       if (object) {
         object.visible = false;
       }
     },
-    [sceneComposerId],
+    [getObject3DBySceneNodeRef],
   );
 
   const remove = useCallback(
     (objectRef: string) => {
-      const removeSceneNode = useStore(sceneComposerId).getState().removeSceneNode;
       removeSceneNode(objectRef);
     },
-    [sceneComposerId],
+    [removeSceneNode],
   );
 
   return (
     <DndProvider backend={HTML5Backend}>
       <Context.Provider
         value={{
-          rootNodes: rootNodes.map((node) => toSceneHeirarchyNode(node, !isEmpty(searchTerms))),
+          rootNodes: rootNodes.map((item) => toSceneHeirarchyNode(item, item.childRefs.length > 0)),
```

Comment:
the hideChild flag is to hide child of the nodes found from search, so we only display a flat list of matching nodes. this change will break it

Reply:
Local testing shows that search is working properly.

Reply:
you should only see the nodes matching the search, not their children (unless the children matches to)

Reply:
Current state is broken. This branch still displays children, however, it also restores the ability to search at all.